src/RichText.js

```diff
@@ -0,0 +1,156 @@
+import {
+    Editor,
+    Modifier,
+    ContentState,
+    convertFromHTML,
+    DefaultDraftBlockRenderMap,
+    DefaultDraftInlineStyle,
+    CompositeDecorator
+} from 'draft-js';
+import * as sdk from  './index';
+
+const BLOCK_RENDER_MAP = DefaultDraftBlockRenderMap.set('unstyled', {
+    element: 'p' // draft uses <div> by default which we don't really like, so we're using <p>
+});
+
+const STYLES = {
+    BOLD: 'strong',
+    CODE: 'code',
+    ITALIC: 'em',
+    STRIKETHROUGH: 's',
+    UNDERLINE: 'u'
+};
+
+const MARKDOWN_REGEX = {
+    LINK: /(?:\[([^\]]+)\]\(([^\)]+)\))|\<(\w+:\/\/[^\>]+)\>/g,
```

Comment:
I know, I was complaining about it on #vector-dev too, but there's very
little that can be done here.

On Tue, Jun 14, 2016 at 12:13 AM David Baker <notifications@github.com>
wrote:

> In src/RichText.js
> <https://github.com/matrix-org/matrix-react-sdk/pull/292#discussion_r66845074>
> :
>
> > +import * as sdk from  './index';
> > +
> > +const BLOCK_RENDER_MAP = DefaultDraftBlockRenderMap.set('unstyled', {
> > +    element: 'p' // draft uses <div> by default which we don't really like, so we're using <p>
> > +});
> > +
> > +const STYLES = {
> > +    BOLD: 'strong',
> > +    CODE: 'code',
> > +    ITALIC: 'em',
> > +    STRIKETHROUGH: 's',
> > +    UNDERLINE: 'u'
> > +};
> > +
> > +const MARKDOWN_REGEX = {
> > +    LINK: /(?:\[([^\]]+)\]\(([^\)]+)\))|\<(\w+:\/\/[^\>]+)\>/g,
>
> Ah right, I see. Still doesn't feel great to have two different types of
> parsing for the markdown, but if it's not on the critical message path then
> probably ok.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/matrix-org/matrix-react-sdk/pull/292/files/df69d1de44810d2c6d8a338be33b73ee8829d177#r66845074>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAOlRHcbqZZTJzWOzcFdS4A5Xabm-zmlks5qLaTZgaJpZM4IoK58>
> .
>